### PR TITLE
[kube-prometheus-stack] Bump kube-state-metrics dependency version

### PR DESCRIPTION
**Pull request by locmai**
#### What this PR does / why we need it:

Bump the dependency chart to the latest 3.5.x version.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 18.1.0
+version: 18.1.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:21]:
```suggestion
version: 18.1.2
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (locmai) [reply charts/kube-prometheus-stack/Chart.yaml]: Bump to 19.0.1 since someone had bump the major first :D

**Comment by locmai** [comment]
@monotek Hi, I have updated the PR, could you help me check one more time :D

**Review by bismarck — APPROVED** [review]
lgtm

**Pull request merged**